File: mobile/lib/models/action_report.dart
class ActionReport {
  final String id;
  final String? type;
  final String title;
  final String description;
  final String body;
  final String? evidenceId;
  final String? characterId;
  final DateTime timestamp;

  const ActionReport({
    required this.id,
    this.type,
    required this.title,
    required this.description,
    required this.body,
    this.evidenceId,
    this.characterId,
    required this.timestamp,
  });

  factory ActionReport.fromJson(Map<String, dynamic> json) {
    return ActionReport(
      id: json['id'] as String,
      type: json['type'] as String?,
      title: json['title'] as String,
      description: json['description'] as String,
      body: json['body'] as String,
      evidenceId: json['evidence_id'] as String?,
      characterId: json['character_id'] as String?,
      timestamp: DateTime.parse(json['timestamp'] as String),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      if (type != null) 'type': type,
      'title': title,
      'description': description,
      'body': body,
      if (evidenceId != null) 'evidence_id': evidenceId,
      if (characterId != null) 'character_id': characterId,
      'timestamp': timestamp.toIso8601String(),
    };
  }
}

File: backend/internal/interfaces/http/dtos/action_report.go
package dtos

import (
	"time"

	"github.com/artcodefun/detective-game/backend/internal/application/ports"
	"github.com/artcodefun/detective-game/backend/internal/application/readmodels"
	"github.com/artcodefun/detective-game/backend/internal/domain"
	"github.com/google/uuid"
)

type ActionReport struct {
	ID          uuid.UUID  `json:"id"`
	Type        string     `json:"type"`
	Title       string     `json:"title"`
	Description string     `json:"description"`
	Body        string     `json:"body"`
	EvidenceID  *uuid.UUID `json:"evidence_id,omitempty"`
	CharacterID *uuid.UUID `json:"character_id,omitempty"`
	Timestamp   time.Time  `json:"timestamp"`
}

func ActionReportFromReadModel(report *readmodels.ActionReport, translator ports.Translator, locale domain.Locale) ActionReport {
	return ActionReport{ID: report.ID, Type: report.Type, Title: translator.Translate(locale, report.Title), Description: report.Description, Body: report.Body, EvidenceID: report.EvidenceID, CharacterID: report.CharacterID, Timestamp: report.Timestamp}
}

func ActionReportsFromReadModels(reports []*readmodels.ActionReport, translator ports.Translator, locale domain.Locale) []ActionReport {
	items := make([]ActionReport, 0, len(reports))
	for _, report := range reports {
		items = append(items, ActionReportFromReadModel(report, translator, locale))
	}
	return items
}
